src/librustc_traits/dropck_outlives.rs

```diff
@@ -193,14 +193,36 @@ fn dtorck_constraint_for_ty<'a, 'gcx, 'tcx>(
             .map(|ty| dtorck_constraint_for_ty(tcx, span, for_ty, depth + 1, ty))
             .collect(),
 
-        ty::TyGenerator(def_id, substs, _) => {
-            // Note that the interior types are ignored here.
-            // Any type reachable inside the interior must also be reachable
-            // through the upvars.
```

Comment:
This comment should be kept in some form. It serves as justification for why we can get away with only reporting the upvars instead of actually calculating all the types which may get dropped inside.

Reply:
(okay, addressed this in followup commit)